### ISSUE TITLE
don't split lists if it's not needed

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/dart_style#146]
Consider this example:
```dart
main() {
  foo([1, 2, 4, 5, 6, 7, 8, 20], argument, argument, argument, argument, argument);
}
```
dartformat produces:

```dart
main() {
  foo([
    1,
    2,
    4,
    5,
    6,
    7,
    8,
    20
  ], argument, argument, argument, argument, argument);
}
```

I rather have it produce:
```dart
main() {
  foo([1, 2, 4, 5, 6, 7, 8, 20], argument, argument, argument, argument,
      argument);
}
```

**Comment by sigmundch** [comment]
Example CL where this occurs: https://codereview.chromium.org/840133003/diff/1/test/core_matchers_test.dart?context=10&column_width=80&tab_spaces=8

**Comment by mezoni** [comment]
I know this is possible to implement a slightly more intellectual formatter of the lists.
Rules from this intellectual formatter can be applied to other situations.
Eg., When formatting the parameters (As you knowe, the parameters is a list of parameter).

Of course, this intellectual formatter will not produce results in such manner.
Eg.
var list = [
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0,
0
}

It can be in condition to work with more complex data such as a multidimensional arrays (the inner lists).

So far, I do not plan to use this software due the lack of good tecniques and alrorithms.

So far, I do not plan to use this software due the lack of good techniques and algorithms.

P.S.
But I could help write a such format of lists, of course, if you agree with me that current results of the list formatter are even a less than a primitive.

**Comment by stevenroose** [comment]
+1. This is annoying and not useful.

**Comment by munificent** [comment]
Fixed by 346ad0736b5aff21ce5751c818bd4ccfc02d5885!
